### PR TITLE
Support configuring AdminClient in KafkaBridge by CR

### PR DESCRIPTION
**Pull request by sknot-rh**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change

- Bugfix
- Enhancement / new feature

### Description

The KafkaBridge CRD should support configuring AminClient used by the Bridge. This is applicable after https://github.com/strimzi/strimzi-kafka-bridge/pull/524 is merged and the new Bridge version is released.
Fixes https://github.com/strimzi/strimzi-kafka-operator/issues/5070

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeAdminClientConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+public class KafkaBridgeAdminClientConfiguration extends AbstractConfiguration {
+
+    private static final List<String> FORBIDDEN_PREFIXES;
+    private static final List<String> FORBIDDEN_PREFIX_EXCEPTIONS;
+    private static final Map<String, String> DEFAULTS;
+
+    static {
+        FORBIDDEN_PREFIXES = AbstractConfiguration.splitPrefixesToList(KafkaBridgeAdminClientSpec.FORBIDDEN_PREFIXES);
+        FORBIDDEN_PREFIX_EXCEPTIONS = AbstractConfiguration.splitPrefixesToList(KafkaBridgeConsumerSpec.FORBIDDEN_PREFIX_EXCEPTIONS);
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeAdminClientConfiguration.java:26]: Should this really link to `KafkaBridgeConsumerSpec.FORBIDDEN_PREFIX_EXCEPTIONS`?

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaBridgeSpec.java [hunk 2]
```diff
@@ -135,6 +136,16 @@ public void setBootstrapServers(String bootstrapServers) {
         this.bootstrapServers = bootstrapServers;
     }
 
+    @Description("Kafka AdminClient related configuration.")
+    @JsonInclude(value = JsonInclude.Include.NON_NULL)
+    public KafkaBridgeAdminClientSpec getAdminClient() {
+        return adminClient;
+    }
+
+    public void setConsumer(KafkaBridgeAdminClientSpec adminClient) {
```
Comment (samuel-hawker) [inline api/src/main/java/io/strimzi/api/kafka/model/KafkaBridgeSpec.java:145]: Why is this not
```suggestion
    public void setAdminClient(KafkaBridgeAdminClientSpec adminClient) {
```

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeAdminClientConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+package io.strimzi.operator.cluster.model;
+
+import io.strimzi.api.kafka.model.KafkaBridgeAdminClientSpec;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class for handling Kafka Bridge consumer configuration passed by the user
```
Comment (samuel-hawker) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeAdminClientConfiguration.java:15]:
```suggestion
 * Class for handling Kafka Bridge admin client configuration passed by the user
```
?

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaBridgeAdminClientSpec.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode(callSuper = true)
+public class KafkaBridgeAdminClientSpec extends KafkaBridgeClientSpec {
+    private static final long serialVersionUID = 1L;
+
+    public static final String FORBIDDEN_PREFIXES = "ssl., bootstrap.servers, group.id, sasl., security.";
```
Comment (scholzj) [inline api/src/main/java/io/strimzi/api/kafka/model/KafkaBridgeAdminClientSpec.java:23]: I think the `group.id` is not needed here.

**Review by scholzj — APPROVED** [review]
One more nit. But LGTM otherwise.

**Review by ppatierno — APPROVED** [review]

**Comment by scholzj** [comment]
@sknot-rh Does this have to wait for Bridge 0.20 release? Or can it be merged right away? Do you know?

**Comment by scholzj** [comment]
Sorry, missed it in the description.

**Comment by sknot-rh** [comment]
Can be merged after https://github.com/strimzi/strimzi-kafka-operator/pull/5120

**Comment by scholzj** [comment]
@sknot-rh #5120 is merged now. so you should be able to rebase this and we should be able to proceed with it.

**Pull request merged**